### PR TITLE
Feature/cloud backup new

### PR DESCRIPTION
**Pull request by antonijzelinskij**
#86944gdfq
image

### CODE REVIEW

**Comment by stepanLav** [comment]
Task linked: [CU-86944gdfq Android. Create Wallet. Your Wallet is Ready screen. User does not have existing backups. ](https://app.clickup.com/t/86944gdfq)

File: common/src/main/java/io/novafoundation/nova/common/utils/KotlinExt.kt [hunk 1]
```diff
@@ -48,6 +48,12 @@ inline fun <T, R> Result<T>.flatMap(transform: (T) -> Result<R>): Result<R> {
     )
 }
 
+inline fun <R> Result<R>.finally(transform: () -> Unit): Result<R> {
+    onSuccess { transform() }
+    onFailure { transform() }
```
Comment (valentunn) [inline common/src/main/java/io/novafoundation/nova/common/utils/KotlinExt.kt:53]: Result is not lazy so you can just call transform() straight away

File: feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/mnemonic/backup/BackupMnemonicFragment.kt [hunk 2]
```diff
@@ -40,9 +47,30 @@ class BackupMnemonicFragment : BaseFragment<BackupMnemonicViewModel>() {
...
     }
 
+    private fun buildConditions() {
+        backupMnemonicCondition1.text = buildCondition(R.string.backup_mnemonic_condition_1, R.string.backup_mnemonic_condition_1_highlight)
+        backupMnemonicCondition2.text = buildCondition(R.string.backup_mnemonic_condition_2, R.string.backup_mnemonic_condition_2_highlight)
+        backupMnemonicCondition3.text = buildCondition(R.string.backup_mnemonic_condition_3, R.string.backup_mnemonic_condition_3_highlight)
+
+        backupMnemonicCondition1.setOnCheckedChangeListener { _, isChecked -> viewModel.conditionClicked(0, isChecked) }
+        backupMnemonicCondition2.setOnCheckedChangeListener { _, isChecked -> viewModel.conditionClicked(1, isChecked) }
+        backupMnemonicCondition3.setOnCheckedChangeListener { _, isChecked -> viewModel.conditionClicked(2, isChecked) }
```
Comment (valentunn) [inline feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/mnemonic/backup/BackupMnemonicFragment.kt:63]: Magic constants here (0, 1, 2)

File: feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/mnemonic/backup/BackupMnemonicViewModel.kt [hunk 3]
```diff
@@ -52,22 +59,18 @@ class BackupMnemonicViewModel(
...
 
-    val continueText = flowOf {
-        val stringRes = when (payload) {
-            is BackupMnemonicPayload.Confirm -> R.string.account_confirm_mnemonic
-            is BackupMnemonicPayload.Create -> R.string.common_continue
+    val continueButtonState = combine(mnemonicWasShown, conditionsState) { mnemonicShown, conditions ->
+        val allConditionsSelected = conditions.values.size == 3 && conditions.values.all { it }
+        when {
+            mnemonicShown && allConditionsSelected -> DescriptiveButtonState.Enabled(resourceManager.getString(R.string.common_confirm))
+            else -> DescriptiveButtonState.Disabled(resourceManager.getString(R.string.backup_mnemonic_disabled_button))
```
Comment (valentunn) [inline feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/mnemonic/backup/BackupMnemonicViewModel.kt:72]: Looks like we show "Agree with conditions" even when all conditions are checked but mnemonic is not shown. Not sure if its expected or not

**Review by valentunn — APPROVED** [review]

**Pull request merged**